packages/plugin-heading/Heading.ts

```diff
@@ -17,4 +55,29 @@ export class Heading extends JWPlugin {
             return [context, parsingMap];
         }
     }
+    /**
+     * Change the formatting of the nodes in given range to Heading.
+     *
+     * @param params
+     */
+    applyParagraph(params: ParagraphParams): void {
+        let heading = params.nodeCreator(); // todo: make sure to keep attributes
+        if (heading.is(HeadingNode)) {
+            const range = params.range || this.editor.vDocument.selection.range;
+            let nodesToConvert: VNode[];
+            if (range.isCollapsed()) {
+                nodesToConvert = [range.start.parent];
+            } else {
+                const selectedLeaves = range.selectedNodes(node => !node.children.length);
```

Comment:
We have `isLeaf` in `VNode.ts` specifically for these use cases. (Basically we tend to avoid custom predicates as much as possible.) Though `isLeaf` will be `true` if the node only contains markers, while yours will be `false`. Does it matter ?